File: app/src/main/java/org/tasks/activities/DeleteCalendarViewModel.java
package org.tasks.activities;

import org.tasks.caldav.CaldavClient;
import org.tasks.data.CaldavAccount;
import org.tasks.data.CaldavCalendar;
import org.tasks.ui.ActionViewModel;

public class DeleteCalendarViewModel extends ActionViewModel {
  public void deleteCalendar(CaldavClient client, CaldavAccount account, CaldavCalendar calendar) {
    run(() -> client.forCalendar(account, calendar).deleteCollection());
  }
}

File: app/src/main/java/org/tasks/todoist/DeleteCalendarViewModel.kt
package org.tasks.todoist

import dagger.hilt.android.lifecycle.HiltViewModel
import org.tasks.data.entity.CaldavAccount
import org.tasks.data.entity.CaldavCalendar
import org.tasks.ui.ActionViewModel
import javax.inject.Inject

@HiltViewModel
class DeleteCalendarViewModel @Inject constructor(
        private val clientProvider: TodoistClientProvider) : ActionViewModel() {
    suspend fun deleteCalendar(account: CaldavAccount, calendar: CaldavCalendar) {
        run { clientProvider.forAccount(account).deleteCollection(calendar) }
    }
}
